### DISCUSSION TITLE
`serializers.JSONField()` openapi definition should include `additionalProperties: true`

### DISCUSSION THREAD

**Discussion by johnnymetz** [encode/django-rest-framework#8776] [Potential Issue]
**Background**

Looks like `my_field = serializers.JSONField()` currently generates the following openapi spec:

```
my_field:
  type: object
```

It should generate the following spec (or at least provide an option to generate it):

```
my_field:
  type: object
  additionalProperties: true
```

The reason being that the latter is considered a "free-form" object, which can have any key-value pairs (see [here](https://swagger.io/docs/specification/data-models/dictionaries/#free-form)).

The former is generally considered an empty object, see here: https://github.com/drwpow/openapi-typescript/issues/1004#issuecomment-1325370100

**Implementation Notes**

- Looks like this can be implemented in `AutoSchema.map_field()` here: https://github.com/encode/django-rest-framework/blob/041b88f8bbb48d9688ebd5add294eee2dfc93d1c/rest_framework/schemas/openapi.py#L368

Probably just need to include the following snippet inside that method:

```
if isinstance(field, serializers.JSONField):
    return {
        'type': 'object',
        'additionalProperties': 'true'
    }
```

**Comment by auvipy** [comment]
you can send an improvement request or you can also try [drf-spectacular ](https://github.com/tfranzel/drf-spectacular) as default schema generator
